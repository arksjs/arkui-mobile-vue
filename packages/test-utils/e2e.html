<script src="../../node_modules/vue/dist/vue.global.js"></script>
<script src="../../dist/index.js"></script>
<link rel="stylesheet" href="../../dist/index.css" />

<div id="app"></div>
