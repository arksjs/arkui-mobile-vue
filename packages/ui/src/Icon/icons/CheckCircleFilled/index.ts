export { default as default } from './CheckCircleFilled.vue'
