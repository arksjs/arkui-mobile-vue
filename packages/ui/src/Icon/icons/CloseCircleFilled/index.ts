export { default as default } from './CloseCircleFilled.vue'
