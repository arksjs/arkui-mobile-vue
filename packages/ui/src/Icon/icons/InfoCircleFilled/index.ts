export { default as default } from './InfoCircleFilled.vue'
