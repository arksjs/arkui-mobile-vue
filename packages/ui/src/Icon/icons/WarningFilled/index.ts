export { default as default } from './WarningFilled.vue'
