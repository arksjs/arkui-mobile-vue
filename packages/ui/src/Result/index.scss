@import '../style/var.scss';

.ak-result {
  padding: 60px 16px;

  &_header {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-bottom: 32px;
  }

  &_icon {
    --ak-icon-size: 64px;
    --ak-icon-color: #{$primary-color};

    &.type {
      &--success {
        --ak-icon-color: #{$success-color};
      }

      &--warning {
        --ak-icon-color: #{$warning-color};
      }

      &--fail {
        --ak-icon-color: #{$danger-color};
      }
    }
  }

  &_title {
    margin-top: 20px;
    line-height: 31px;
    font-weight: 700;
    font-size: 22px;
    color: $title-color;
    word-break: break-all;
    word-wrap: break-word;
    overflow: hidden;
  }

  &_description {
    margin-top: 16px;
    font-size: 17px;
    line-height: 24px;
    color: $font-color;
    word-break: break-all;
    word-wrap: break-word;
    overflow: hidden;
    text-align: center;
  }

  &_footer {
    margin-top: 32px;
    display: flex;
    flex-direction: column;

    .ak-button + .ak-button {
      margin: 12px 0 0;
    }
  }
}
