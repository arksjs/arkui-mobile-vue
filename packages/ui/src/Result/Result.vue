<template>
  <div class="ta-result" :class="typeClass">
    <div class="ta-result_header">
      <Icon class="ta-result_icon" :class="typeClass" :icon="icon" />
      <div class="ta-result_title" v-if="title">{{ title }}</div>
      <div class="ta-result_description" v-if="description">
        {{ description }}
      </div>
    </div>
    <slot></slot>
    <div class="ta-result_footer">
      <TaButton type="primary" size="large" @click="onConfirmClick">
        {{ confirmText || locale.resultConfirmText }}
      </TaButton>
      <TaButton
        v-if="showBack"
        type="default"
        size="large"
        @click="onCancelClick"
      >
        {{ backText || locale.resultBackText }}
      </TaButton>
    </div>
  </div>
</template>

<script lang="ts">
import { computed, defineComponent, type PropType } from 'vue'
import { Icon } from '../Icon'
import { Button as TaButton } from '../Button'
import {
  createEnumsValidator,
  emitClickValidator,
  type PropsToEmits
} from '../helpers'
import { useLocale } from '../ConfigProvider/context'
import type { ResultEmits, ResultType } from './types'
import InfoCircleFilled from '../Icon/icons/InfoCircleFilled'
import WarningFilled from '../Icon/icons/WarningFilled'
import CheckCircleFilled from '../Icon/icons/CheckCircleFilled'
import CloseCircleFilled from '../Icon/icons/CloseCircleFilled'
import type { IconData } from '../Icon/types'
import { getType, getTypeClass, RESULT_TYPES } from './util'

const iconMap = new Map<ResultType, IconData>([
  ['info', InfoCircleFilled],
  ['warning', WarningFilled],
  ['success', CheckCircleFilled],
  ['fail', CloseCircleFilled]
])

export default defineComponent({
  name: 'ta-result',
  components: { Icon, TaButton },
  props: {
    // 类型
    type: {
      type: String as PropType<ResultType>,
      validator: createEnumsValidator(RESULT_TYPES)
    },
    // 标题
    title: {
      type: String,
      default: ''
    },
    // 描述
    description: {
      type: String,
      default: ''
    },
    // 展示返回按钮
    showBack: {
      type: Boolean,
      default: true
    },
    // 返回文本文本
    backText: {
      type: String
    },
    // 确认文本
    confirmText: {
      type: String
    }
  },
  emits: {
    confirm: emitClickValidator,
    back: emitClickValidator
  } as PropsToEmits<ResultEmits>,
  setup(props, { emit }) {
    const { locale } = useLocale()
    const typeClass = computed(() => getTypeClass(props.type))
    const icon = computed(() => iconMap.get(getType(props.type)) as IconData)

    function onConfirmClick(e: MouseEvent) {
      emit('confirm', e)
    }

    function onCancelClick(e: MouseEvent) {
      emit('back', e)
    }

    return {
      typeClass,
      icon,
      onConfirmClick,
      onCancelClick,
      locale
    }
  }
})
</script>
