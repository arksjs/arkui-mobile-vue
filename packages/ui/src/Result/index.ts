import { withInstall } from '../helpers/with-install'
import Result from './Result.vue'

export { Result }
export const install = withInstall(Result)
export default Result
