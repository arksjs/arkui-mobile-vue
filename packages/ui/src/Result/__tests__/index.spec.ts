import { describe, test, expect } from 'vitest'
import { mount, shallowMount } from '@vue/test-utils'
import { Result } from '@/Result'
import InfoCircleFilled from '@/Icon/icons/InfoCircleFilled'
import WarningFilled from '@/Icon/icons/WarningFilled'
import CheckCircleFilled from '@/Icon/icons/CheckCircleFilled'
import CloseCircleFilled from '@/Icon/icons/CloseCircleFilled'

describe('Result', () => {
  test('snapshot', () => {
    const wrapper = shallowMount(Result, {})
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('type', async () => {
    const wrapper = mount(Result, {
      props: {
        type: 'info'
      }
    })
    expect(wrapper.classes()).toContain('type--info')
    expect(wrapper.findComponent(InfoCircleFilled).exists()).toBeTruthy()

    await wrapper.setProps({
      type: 'warning'
    })
    expect(wrapper.classes()).toContain('type--warning')
    expect(wrapper.findComponent(WarningFilled).exists()).toBeTruthy()

    await wrapper.setProps({
      type: 'success'
    })
    expect(wrapper.classes()).toContain('type--success')
    expect(wrapper.findComponent(CheckCircleFilled).exists()).toBeTruthy()

    await wrapper.setProps({
      type: 'fail'
    })
    expect(wrapper.classes()).toContain('type--fail')
    expect(wrapper.findComponent(CloseCircleFilled).exists()).toBeTruthy()
  })

  const title = 'title'
  const description = 'description'
  test('title & description', () => {
    const wrapper = shallowMount(Result, {
      props: {
        title,
        description
      }
    })
    expect(wrapper.find('.ta-result_title').text()).toBe(title)
    expect(wrapper.find('.ta-result_description').text()).toBe(description)
  })

  test('showBack', () => {
    const wrapper = mount(Result, {
      props: {
        showBack: true
      }
    })
    expect(wrapper.findAllComponents('.ta-button').length).toBe(2)
  })

  const confirmText = 'confirmText'
  const backText = 'backText'

  test('confirmText & backText', () => {
    const wrapper = mount(Result, {
      props: {
        showBack: true,
        confirmText,
        backText
      }
    })
    expect(
      wrapper
        .findAllComponents('.ta-button')
        .map(v => v.find('.ta-button').text())
    ).toEqual([confirmText, backText])
  })
})
