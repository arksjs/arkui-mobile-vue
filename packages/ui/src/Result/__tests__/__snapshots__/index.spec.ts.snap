// Vitest Snapshot v1

exports[`Result > snapshot 1`] = `
"<div class=\\"ak-result type--info\\">
  <div class=\\"ak-result_header\\">
    <icon-stub icon=\\"[object Object]\\" spin=\\"false\\" size=\\"0\\" width=\\"0\\" height=\\"0\\" class=\\"ak-result_icon type--info\\"></icon-stub>
    <!--v-if-->
    <!--v-if-->
  </div>
  <div class=\\"ak-result_footer\\">
    <ak-button-stub type=\\"primary\\" loading=\\"false\\" disabled=\\"false\\" ghost=\\"false\\" transparent=\\"false\\"></ak-button-stub>
    <ak-button-stub type=\\"default\\" loading=\\"false\\" disabled=\\"false\\" ghost=\\"false\\" transparent=\\"false\\"></ak-button-stub>
  </div>
</div>"
`;
